.gitlab/ci/.gitlab-ci.sdk-nightly.yml

```diff
@@ -235,26 +345,183 @@ demisto-sdk-nightly:run-end-to-end-tests:
+    - cat "$ARTIFACTS_FOLDER/modeling_rules_to_test.txt"
+    - section_end "Update packs to install and modeling rules to test."
+
+    - section_start "Create Artifacts instance directory" --collapsed
     - |
-      mkdir $ARTIFACTS_FOLDER/demisto-sdk
-      git clone -b ${SDK_REF} --single-branch --depth 1 https://github.com/demisto/demisto-sdk.git $ARTIFACTS_FOLDER/demisto-sdk
+      echo "Creating Artifacts instance directory: ${ARTIFACTS_FOLDER_INSTANCE}"
+      [ ! -d "${ARTIFACTS_FOLDER_INSTANCE}" ] && mkdir -p "${ARTIFACTS_FOLDER_INSTANCE}"
+      echo "${INSTANCE_ROLE}" > "${ARTIFACTS_FOLDER_INSTANCE}/instance_role.txt"
+    - section_end "Create Artifacts instance directory"
 
-      TUNNEL_PORT=$(cat $ARTIFACTS_FOLDER/env_results.json | jq "map(select(.Role == \"$INSTANCE_ROLE\")) | .[].TunnelPort")
-      export DEMISTO_BASE_URL="https://$(cat $ARTIFACTS_FOLDER/env_results.json | jq -r '.[0].InstanceDNS')"
-      echo "Server URL: $DEMISTO_BASE_URL"
-      unset DEMISTO_API_KEY # a temporary fix 
-      python3 -m pytest $ARTIFACTS_FOLDER/demisto-sdk/tests_end_to_end/ || EXIT_CODE=$?
+    - !reference [.download-demisto-conf]
+    - !reference [.secrets-fetch]
+    - !reference [.lock-machine]
+
+    - !reference [.export_cloud_machine_constants]
+    - echo "Server URL $DEMISTO_BASE_URL"
+    - echo "XSIAM_AUTH_ID $XSIAM_AUTH_ID"
+
+    - section_start "Copy env results to artifacts folder" --collapsed
+    - |
+      # workaround for the hard-coded value in the sdk
+      cp "${ARTIFACTS_FOLDER}/filter_file.txt" "./artifacts/filter_file.txt"
+    - section_end "Copy env results to artifacts folder"
+
+    - section_start "Test Modeling Rules" --collapsed
+    - demisto-sdk upload -z -x -i Packs/SemperisDSP --insecure
+    - ./Tests/scripts/test_modeling_rules.sh || EXIT_CODE=$?
+    - section_end "Test Modeling Rules"
+
+    - section_start "Create Entities and Upload" --collapsed
+    - python3 Tests/sdknightly/create_entities_for_nightly_sdk.py HelloWorld --artifacts-folder $ARTIFACTS_FOLDER
+    - demisto-sdk upload -i Packs/HelloWorld --insecure -z -x
+    - section_end "Create Entities and Upload"
+
+    - section_start "Download Entities" --collapsed
+    - demisto-sdk download -lf --insecure
+    - demisto-sdk download -o Packs/HelloWorld/ -i CommonServerUserPowerShell --insecure -f
+    - section_end "Download Entities"
+
+    - section_start "Install Packs" --collapsed
+    - cat "$ARTIFACTS_FOLDER/filter_file.txt" 
+    - ./Tests/scripts/install_content_and_test_integrations.sh || EXIT_CODE=$?
+    - cp -f "${ARTIFACTS_FOLDER}/conf.json" Tests/conf.json
+    - section_end "Install Packs"
+
+    - section_start "Wait Until Server Ready"  --collapsed
+    - echo Going to sleep for 15 minutes to allow server finish indexing
+    - sleep-with-progress 900 30 "Sleeping... " 150
+    - echo "Done sleeping!"
+    - section_end "Wait Until Server Ready"
```

Comment:
not needed for cloud machine IMO, but double check

Reply:
Will remove it and let you know

Reply:
Checked with @daryakoval, and there is no need to wait on cloud machines, as the requests are done synchronically.